lib/pure/concurrency/cpuinfo.nim

```diff
@@ -131,178 +131,697 @@ when onX86:
 
 when onX86 or defined(nimdoc):
   type
-    X86Feature* {.pure.} = enum
-      HypervisorPresence, SimultaneousMultithreading, Vmx, X87fpu, Mmx, MmxExt,
-      F3DNow, F3DNowExt, F3DNowPf, Sse, Sse2, Sse3, Ssse3, Sse4a, Sse41, Sse42,
-      Avx, Avx2, Avx512f, Avx512dq, Avx512ifma, Avx512pf, Avx512er, Avx512cd,
-      Avx512bw, Avx512vl, Avx512vbmi, Avx512vbmi2, Avx512vpopcntdq, Avx512vnni,
-      Avx512vnniw4, Avx512fmaps4, Avx512bitalg, Rdrand, Rdseed, MovBigEndian,
-      Popcnt, Fma3, Fma4, Cas8B, Cas16B, Abm, Bmi1, Bmi2, TsxHle, TsxRtm, Adx,
-      Sgx, Gfni, Aes, Vaes, Vpclmulqdq, Pclmulqdq, NxBit, Float16c, Ssbd,
-      SpecCtrl, Stibp, Sha, Clflush, ClflushOpt, Clwriteback, PrefetchWT1, Mpx
-
-when onArm or defined(nimdoc):
-  type
-    ArmFeature* {.pure.} = enum
-      Neon, Vfp
-
-type CpuFeature* = (when onX86: X86Feature elif onArm: ArmFeature)
+    X86Feature {.pure.} = enum
```

Comment:
I would put everything under the `when onX86 or defined(nimdoc):`
in a separate `cpuinfo_x86` and just do:

```nim
when onX86 or defined(nimdoc):
  include ./cpuinfo_x86
```